planning/behavior_path_planner/src/scene_module/lane_change/util.cpp

```diff
@@ -375,30 +375,22 @@ bool hasEnoughDistance(
     return false;
   }
 
-  if (
-    lane_change_total_distance + lane_change_required_distance >
-    util::getDistanceToNextIntersection(current_pose, current_lanes)) {
-    return false;
-  }
-
   if (
     route_handler.isInGoalRouteSection(current_lanes.back()) &&
     lane_change_total_distance + lane_change_required_distance >
       util::getSignedDistance(current_pose, goal_pose, current_lanes)) {
     return false;
   }
 
-  if (
-    lane_change_total_distance + lane_change_required_distance >
-    util::getDistanceToCrosswalk(current_pose, current_lanes, *overall_graphs)) {
-    return false;
-  }
-
   // return is there is no target lanes. Else continue checking
   if (target_lanes.empty()) {
     return true;
   }
 
+  if (lane_change_total_distance > util::getDistanceToEndOfLane(current_pose, target_lanes)) {
+    return false;
+  }
```

Comment:
@zulfaqar-azmi-t4
This change will fix the problem mentioned in following: 
https://github.com/autowarefoundation/autoware.universe/pull/2442#issuecomment-1399724526